packages/ui/ViewManager/View.js

```diff
@@ -163,6 +163,12 @@ class View extends Component {
 
 	componentDidUpdate (prevProps) {
 		this.changeDirection = this.shouldChangeDirection(prevProps, this.props);
+
+		// this is needed to avoid setting enteringProp to true in the render method when noAnimation is true
+		if (prevProps.noAnimation !== this.props.noAnimation && this.state.entering) {
```

Comment:
We could use `componentDidEnter` instead of this lifecycle.
Please check out my branch https://github.com/enactjs/enact/compare/develop...feature/PLAT-141617-mk.

Reply:
👍